lib/sync.js

```diff
@@ -466,7 +469,7 @@ SyncApi.prototype._sync = function(syncOptions, attempt) {
                 var accountDataEvents = self._mapSyncEventsFormat(joinObj.account_data);
 
                 // we do this first so it's correct when any of the events fire
-                room.unread_notification_count = joinObj.unread_notification_count;
+                room.unread_notifications = joinObj.unread_notifications || {};
```

Comment:
Done. nb. I've included 'counts' in the name as it is the count of unread notifs, not the actual notifs. Also I've made the type of count a string param rather than a boolean as just having a boolean makes backwards compat more awkward if we add more types.